core/src/spotify_id.rs

```
@@ -13,7 +14,7 @@ const BASE62_DIGITS: &'static [u8] = b"0123456789abcdefghijklmnopqrstuvwxyzABCDE
 const BASE16_DIGITS: &'static [u8] = b"0123456789abcdef";
 
 impl SpotifyId {
-    pub fn from_base16(id: &str) -> SpotifyId {
+    pub fn from_base16(id: &str) -> Result<SpotifyId> {
         assert!(id.is_ascii());
```

That should return an error

---

And actually this check isn't really needed at all

---

So do you want an error or to delete that assert?

---

Ah right, got it, Github collapsing the text made this ambiguous at first glance.